src/RA_Dlg_Login.cpp

```diff
@@ -5,118 +5,150 @@
-
-                            RAUsers::LocalUser().ProcessSuccessfulLogin(sUser, sToken, nPoints, nUnreadMessages, bRememberLogin);
-
-                            sResponse = "Logged in as " + sUser + ".";
-                            sResponseTitle = "Logged in Successfully!";
-
-                            //g_PopupWindows.AchievementPopups().SuppressNextDeltaUpdate();
-                        }
-                        else
-                        {
-                            sResponse = std::string("Failed!\r\n"
-                                "Response From Server:\r\n") +
-                                doc["Error"].GetString();
-                            sResponseTitle = "Error logging in!";
-                        }
-
-                        MessageBox(hDlg, NativeStr(sResponse).c_str(), NativeStr(sResponseTitle).c_str(), MB_OK);
-
-                        //	If we are now logged in
-                        if (RAUsers::LocalUser().IsLoggedIn())
-                        {
-                            //	Close this dialog
-                            EndDialog(hDlg, TRUE);
-                        }
-
-                        return TRUE;
+                        MessageBox(hDlg, NativeStr(std::string("Server error: ") +
+                                   std::string(doc["Error"].GetString())).c_str(), TEXT("Error!"), MB_OK);
                     }
                     else
                     {
-                        if (!doc.HasParseError() && doc.HasMember("Error"))
-                        {
-                            MessageBox(hDlg, NativeStr(std::string("Server error: ") + std::string(doc["Error"].GetString())).c_str(), TEXT("Error!"), MB_OK);
-                        }
-                        else
-                        {
-                            MessageBox(hDlg, TEXT("Unknown error connecting... please try again!"), TEXT("Error!"), MB_OK);
-                        }
-
-                        return TRUE;	//	==Handled
+                        MessageBox(hDlg, TEXT("Unknown error connecting... please try again!"), TEXT("Error!"),
+                                   MB_OK);
                     }
-                }
-                break;
 
-                case IDCANCEL:
-                    EndDialog(hDlg, TRUE);
-                    return TRUE;
+                    break;	//	==Handled
+                }
             }
             break;
 
-        case WM_CLOSE:
-            EndDialog(hDlg, TRUE);
-            return TRUE;
+            case IDCANCEL:
+                EndDialog(hDlg, IDCANCEL);
+        }
     }
+};
 
+INT_PTR Dlg_Login::DoModal()
+{
+    return DialogBoxParam(::GetModuleHandle(_T("RA_Integration.dll")), MAKEINTRESOURCE(IDD_RA_LOGIN),
+                          ::GetAncestor(::GetActiveWindow(), GA_ROOT), Dlg_LoginProc,
+                          reinterpret_cast<LPARAM>(this));
+}
```

Comment:
Figured this was one way to limit globals, so far what I've noticed is the only difference between these "DoModal" is the resource ID, and maybe the the parent window (Memory Bookmarker).

`::GetAncestor(::GetActiveWindow(), GA_ROOT)` is `g_RAMainWnd`.
It was called this way because ::GetActiveWindow() by itself might grab the wrong handle.

`reinterpret_cast<LPARAM>(this)` gives a default WindowClass, might not seem too useful here but when these dialogs get abstracted it will be required because the `Message Queue` will need to differentiate between the Dialogs or it will call the wrong procedure.